cmd/kubeadm/app/cmd/init.go

```diff
@@ -397,17 +396,12 @@ func (i *Init) Run(out io.Writer) error {
 	fmt.Println("[init] this might take a minute or longer if the control plane images have to be pulled")
 
 	if err := waitForKubeletAndFunc(waiter, waiter.WaitForAPI); err != nil {
-		ctx := map[string]string{
-			"Error":                  fmt.Sprintf("%v", err),
-			"APIServerImage":         images.GetKubeControlPlaneImage(kubeadmconstants.KubeAPIServer, i.cfg),
-			"ControllerManagerImage": images.GetKubeControlPlaneImage(kubeadmconstants.KubeControllerManager, i.cfg),
-			"SchedulerImage":         images.GetKubeControlPlaneImage(kubeadmconstants.KubeScheduler, i.cfg),
-		}
-		// Set .EtcdImage conditionally
-		if i.cfg.Etcd.Local != nil {
-			ctx["EtcdImage"] = fmt.Sprintf("				- %s", images.GetEtcdImage(i.cfg))
-		} else {
-			ctx["EtcdImage"] = ""
+		ctx := struct {
+			Error  string
+			Images []string
+		}{
+			Error:  fmt.Sprintf("%v", err),
+			Images: images.GetAllImages(i.cfg),
```

Comment:
If this includes images other than the control plan I think this will be more confusing.